r/R/dataset-scan.R

```diff
@@ -157,13 +157,19 @@ ScannerBuilder <- R6Class("ScannerBuilder", inherit = ArrowObject,
   public = list(
     Project = function(cols) {
       # cols is either a character vector or a named list of Expressions
-      if (!is.character(cols)) {
-        # We don't yet support mutate() on datasets, so this is just a list
-        # of FieldRefs, and we need to back out the field names
-        cols <- get_field_names(cols)
+      if (is.character(cols)) {
+        dataset___ScannerBuilder__ProjectNames(self, cols)
```

Comment:
In what circumstance will `is.character(cols)` be `TRUE`? I've running a bunch of examples and I've never seen the code take this path.

Reply:
Only if you built a `ScannerBuilder` directly and called `Project` on it. All of the dplyr machinery deals with Expressions now.

Reply:
Ok, thanks